src/Weather.java

```diff
@@ -0,0 +1,170 @@
+
+import components.map.Map;
+import components.map.Map1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+import components.utilities.FormatChecker;
+
+/**
+ * @author Haechan Jung
+ *
+ */
+public class Weather<date, condition> {
+    // 1. extend Pair<K,V> - refer to SimplePair<K, V> -- abstract class
+
+    /*
+     * Private members --------------------------------------------------------
+     */
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Map<String, String> weather;
+
+    /**
+     * Creator of initial representation.
+     */
+    private void createNewRep() {
+        this.weather = new Map1L<String, String>();
+    }
+
+    /*
+     * Constructors -----------------------------------------------------------
+     */
+
+    /**
+     * No-argument constructor.
+     */
+    public Weather() {
+        this.createNewRep();
+    }
+
+    /**
+     * Constructor from {@code date} and {@code condition}.
+     *
+     * @param date
+     *            {@code date} to initialize from
+     * @param condition
+     *            {@code condition} to initialize from
+     */
+    public Weather(String date, String condition) {
+        this.createNewRep();
+        this.addWeather(date, condition);
+    }
+
+    /*
+     * Kernel methods ---------------------------------------------------------
+     */
+    /**
+     * Adds the pair ({@code date}, {@code condition}) to this.
+     *
+     * @param key
+     *            the key to be added
+     * @param value
+     *            the associated value to be added
+     * @aliases references {@code key, value}
+     * @updates this
+     * @requires key is not in DOMAIN(this)
+     * @ensures this = #this union {(key, value)}
+     */
+    public final void addWeather(String date, String condition) {
+        assert date != null : "Violation of: date is not null";
+        assert condition != null : "Violation of: condition is not null";
+        assert this
+                .isDateFormatted(date) : "Violation of: date is not formatted";
+        this.weather.add(date, condition);
+    }
+
+    public final Weather<date, condition> removeWeather(String date) {
```

Comment:
This is somewhat confusing to me. Your object is called Weather. How can this object return an instance of itself? I don't think there is anything wrong with this, but I'd need to know why you're returning a Weather object.